app/services/idv/analytics_events_enhancer.rb

```diff
@@ -174,6 +175,14 @@ def pending_profile_idv_level
       user&.respond_to?(:pending_profile) && user&.pending_profile&.idv_level
     end
 
+    def app_identifier
+      sp_session = session&.fetch(:sp, {})
+
+      return if sp_session.blank? || sp_session['request_url'].blank?
+
+      UriService.params(URI(sp_session['request_url']))['app_identifier']
```

Comment:
maybe could streamline this a bit by grabbing the `request_url`? Also `UriService` ends up casting to `URI` inside it, so we can save a step by removing that too just in case

```suggestion
      sp_request_url = session&.dig(:sp, 'request_url')

      return if sp_request_url.blank?

      UriService.params(sp_request_url)['app_identifier']
```

Reply:
i just pushed a commit putting the work in the analytics file -- if i end up going back to this solution i can definitely streamline! thanks.